.claude/skills/ad-layer-visualizer/scripts/render_layer.py

```
@@ -0,0 +1,592 @@
+    lines.append("}")
+    return "\n".join(lines)
+
+
+# ---------------------------------------------------------------------------
+# Rendering
+# ---------------------------------------------------------------------------
+
+
+def render_png(dot_path: str, png_path: str, dpi: int = 150) -> bool:
+    try:
+        result = subprocess.run(
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)